File: src/dotnet/Ansight.Tools.VisualTree/GetOverlayTool.cs
namespace Ansight.Tools.VisualTree;

public sealed class GetOverlayTool : ITool
{
    public string Category => "ui";

    public ToolPolicy Policy => ToolPolicy.Read;

    public string Id => VisualTreeToolIds.GetOverlay;

    public string Name => "Get Overlay";

    public string Description => "Returns metadata and geometry for a live diagnostic overlay.";

    public string Keywords => "ui overlay highlight inspect metadata";

    public ToolSchema ArgumentsSchema => VisualTreeToolSchemas.GetOverlayArguments;

    public ToolSchema ResultSchema => VisualTreeToolSchemas.OverlayResult;

    public Task<ToolResult> Execute(IReadOnlyDictionary<string, string> arguments)
    {
        ArgumentNullException.ThrowIfNull(arguments);
        return VisualTreeSupport.GetOverlayAsync(arguments);
    }
}

File: src/ios/Sources/AnsightToolsVisualTree/GetOverlayTool.swift
import AnsightCore
import Foundation

public final class GetOverlayTool: AnsightTool {
    public init() {}

    public var descriptor: AnsightToolDescriptor {
        AnsightToolDescriptor(
            id: AnsightVisualTreeToolIds.getOverlay,
            name: "Get Overlay",
            description: "Returns metadata and geometry for a live diagnostic overlay.",
            category: "ui",
            policy: .read,
            keywords: "ui overlay highlight inspect metadata",
            argumentsSchema: AnsightVisualTreeToolSchemas.getOverlayArguments,
            resultSchema: AnsightVisualTreeToolSchemas.overlayResult
        )
    }

    public func execute(arguments: [String: String]) throws -> AnsightToolExecutionResult {
        AnsightVisualTreeOverlaySupport.getOverlay(arguments: arguments)
    }
}
